### PR TITLE
fix myapp-02 example build and run instructions

### LINKED ISSUE #20: examples/myapp-02 'cargo build' fails
Running `cargo build` in `examples/myapp-02` fails:
```
   Compiling myapp v0.1.0 (/home/dmitris
![Screenshot 2022-01-14 at 13 18 35](https://user-images.githubusercontent.com/31205/149514426-0c779154-177b-4f14-baf3-d3d5d1493c22.png)
![Screenshot 2022-01-14 at 13 18 35](https://user-images.githubusercontent.com/31205/149514485-be4ff808-d901-4ee0-8b87-5d73df53f785.png)
/dev/hack/gh/aya-rs/book/examples/myapp-02/myapp)
error[E0277]: the `?` operator can only be used on `Result`s, not `Option`s, in an async block that returns `Result`
  --> myapp/src/main.rs:31:49
   |
18 |   async fn main() -> Result<(), anyhow::Error> {
   |  ______________________________________________-
19 | |     let path = match env::args().nth(1) {
20 | |         Some(iface) => iface,
21 | |         None => panic!("not path provided"),
...  |
31 | |     let probe: &mut Xdp = bpf.program_mut("xdp")?.try_into()?;
   | |                                                 ^ use `.ok_or(...)?` to provide an error compatible with `Result<(), anyhow::Error>`
...  |
60 | |     Ok::<_, anyhow::Error>(())
61 | | }
   | |_- this function returns a `Result`
   |
   = help: the trait `FromResidual<Option<Infallible>>` is not implemented for `Result<(), anyhow::Error>`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `myapp` due to previous error
```

### PR DESCRIPTION
**Pull request by dmitris**
Fixes #20.

### CODE REVIEW

File: examples/myapp-02/myapp/src/main.rs [hunk 1]
```diff
@@ -28,7 +28,7 @@ async fn main() -> Result<(), anyhow::Error> {
     let data = fs::read(path)?;
     let mut bpf = Bpf::load(&data)?;
 
-    let probe: &mut Xdp = bpf.program_mut("xdp")?.try_into()?;
+    let probe: &mut Xdp = bpf.program_mut("xdp").ok_or(anyhow::anyhow!("failed to load bpf program"))?.try_into()?;
```
Comment (dave-tucker) [inline examples/myapp-02/myapp/src/main.rs:31]:
```suggestion
    let probe: &mut Xdp = bpf.program_mut("xdp").unwrap().try_into()?;
```

File: examples/myapp-03/myapp/src/main.rs [hunk 2]
```diff
@@ -26,7 +26,7 @@ async fn main() -> Result<(), anyhow::Error> {
     let data = fs::read(path)?;
     let mut bpf = Bpf::load(&data)?;
 
-    let probe: &mut Xdp = bpf.program_mut("xdp")?.try_into()?;
+    let probe: &mut Xdp = bpf.program_mut("xdp").ok_or(anyhow::anyhow!("failed to load bpf program"))?.try_into()?;
```
Comment (dave-tucker) [inline examples/myapp-03/myapp/src/main.rs:29]:
```suggestion
    let probe: &mut Xdp = bpf.program_mut("xdp").unwrap().try_into()?;
```

**Review by dave-tucker — CHANGES_REQUESTED** [review]
Please `unwrap()` vs using `ok_or` since the only way you would get an error here is if you had made a typo in the section name

File: examples/myapp-03/myapp/src/main.rs (hunk 2 above)
Comment (dmitris) [reply examples/myapp-03/myapp/src/main.rs]: changed in a18bef3

File: examples/myapp-02/myapp/src/main.rs (hunk 1 above)
Comment (dmitris) [reply examples/myapp-02/myapp/src/main.rs]: changed to `unwrap` in a18bef3

**Comment by dmitris** [comment]
> Please `unwrap()` vs using `ok_or` since the only way you would get an error here is if you had made a typo in the section name

done - changed to:
`let probe: &mut Xdp = bpf.program_mut("xdp").unwrap().try_into()?;`

**Review by dave-tucker — APPROVED** [review]

**Pull request merged**